faraday/uart/layer_2_protocol.py

```diff
@@ -15,19 +15,28 @@
 import time
 import struct
 import logging
+import sys
 
 # Get configured logger
 logger = logging.getLogger('UARTStack')
 
 
 class layer_2_object(object):
     def __init__(self, port, baud, timeout):
-        self.serial_physical_obj = layer_2_protocol(port, baud, timeout)
+        # Start up serial object and if it fails exit the program
+        try:
+            self.serial_physical_obj = layer_2_protocol(port, baud, timeout)
```

Comment:
Move this try/catch block into the layer_2_protocol constructor so it is clear what SerialException is being caught (the one that might be thrown by the serial.Serial constructor).